parsl/executors/high_throughput/interchange.py

```diff
@@ -217,79 +217,68 @@ def _send_monitoring_info(self, monitoring_radio: Optional[MonitoringRadioSender
         """
-        logger.debug("Command Server Starting")
-
-        if self.hub_address is not None and self.hub_zmq_port is not None:
-            logger.debug("Creating monitoring radio to %s:%s", self.hub_address, self.hub_zmq_port)
-            monitoring_radio = ZMQRadioSender(self.hub_address, self.hub_zmq_port)
-        else:
-            monitoring_radio = None
+        logger.debug("entering command_server section")
 
         reply: Any  # the type of reply depends on the command_req received (aka this needs dependent types...)
 
-        while True:
-            try:
-                command_req = self.command_channel.recv_pyobj()
-                logger.debug("Received command request: {}".format(command_req))
-                if command_req == "CONNECTED_BLOCKS":
-                    reply = self.connected_block_history
-
-                elif command_req == "WORKERS":
-                    num_workers = 0
-                    for manager in self._ready_managers.values():
-                        num_workers += manager['worker_count']
-                    reply = num_workers
-
-                elif command_req == "MANAGERS":
-                    reply = []
-                    for manager_id in self._ready_managers:
-                        m = self._ready_managers[manager_id]
-                        idle_since = m['idle_since']
-                        if idle_since is not None:
-                            idle_duration = time.time() - idle_since
-                        else:
-                            idle_duration = 0.0
-                        resp = {'manager': manager_id.decode('utf-8'),
-                                'block_id': m['block_id'],
-                                'worker_count': m['worker_count'],
-                                'tasks': len(m['tasks']),
-                                'idle_duration': idle_duration,
-                                'active': m['active'],
-                                'parsl_version': m['parsl_version'],
-                                'python_version': m['python_version'],
-                                'draining': m['draining']}
-                        reply.append(resp)
-
-                elif command_req.startswith("HOLD_WORKER"):
-                    cmd, s_manager = command_req.split(';')
-                    manager_id = s_manager.encode('utf-8')
-                    logger.info("Received HOLD_WORKER for {!r}".format(manager_id))
-                    if manager_id in self._ready_managers:
-                        m = self._ready_managers[manager_id]
-                        m['active'] = False
-                        self._send_monitoring_info(monitoring_radio, m)
+        if self.command_channel in self.socks and self.socks[self.command_channel] == zmq.POLLIN:
+
+            command_req = self.command_channel.recv_pyobj()
+            logger.debug("Received command request: {}".format(command_req))
```

Comment:
The previous iteration had this in a try-except; is `zmq.Again` no longer a concern?  (Perhaps it never was?)

Of course, the previous iteration also wrapped _everything_ in the try-except, so :vomiting_face: to that.

Reply:
this was part of polling-to-exit handling which was mostly removed by #2429 and #2438 - it could have removed this try/except series, I think, but did not.